server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala

```diff
@@ -0,0 +1,101 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.mail
+
+import org.apache.james.core.Username
+import org.apache.james.jmap.model.UnsignedInt
+import org.apache.james.mailbox.Role
+import org.apache.james.mailbox.model.MailboxId
+
+final case class MailboxName(name: String) {
+  require(!name.isEmpty, "'name' is mandatory")
+}
+
+final case class MailboxRights(mayReadItems: Boolean,
+                               mayAddItems: Boolean,
+                               mayRemoveItems: Boolean,
+                               maySetSeen: Boolean,
+                               maySetKeywords: Boolean,
+                               mayCreateChild: Boolean,
+                               mayRename: Boolean,
+                               mayDelete: Boolean,
+                               maySubmit: Boolean)
+
+object MailboxNamespace {
+  def delegated(owner: Username) = DelegatedNamespace(owner)
+
+  def personal = PersonalNamespace
+}
+
+sealed trait MailboxNamespace {
+  def `type`: String
+  def owner: Option[Username]
+}
+
+case object PersonalNamespace extends MailboxNamespace {
+  override def `type`: String = "Personal"
+
+  override def owner: Option[Username] = None
+}
+
+case class DelegatedNamespace(user: Username) extends MailboxNamespace {
+  override val `type`: String = "Delegated"
+
+  override val owner: Option[Username] = Some(user)
+}
+
+object SortOrder {
+  private val defaultSortOrders = Map(
+      Role.INBOX -> SortOrder(10L),
+      Role.ARCHIVE -> SortOrder(20L),
+      Role.DRAFTS -> SortOrder(30L),
+      Role.OUTBOX -> SortOrder(40L),
+      Role.SENT -> SortOrder(50L),
+      Role.TRASH -> SortOrder(60L),
+      Role.SPAM -> SortOrder(70L),
+      Role.TEMPLATES -> SortOrder(80L),
+      Role.RESTORED_MESSAGES -> SortOrder(90L))
+    .withDefaultValue( SortOrder(1000L))
+
+  def getSortOrder(role: Role): SortOrder = defaultSortOrders(role)
+}
+
+case class SortOrder private(sortOrder: UnsignedInt) extends Ordered[SortOrder] {
+  override def compare(that: SortOrder): Int = this.sortOrder.compare(that.sortOrder)
+}
+
+case class Mailbox(id: MailboxId,
+                   mailboxName: MailboxName,
+                   parentId: Option[MailboxId],
+                   role: Option[Role],
+                   sortOrder: SortOrder,
+                   totalEmails: UnsignedInt,
+                   unreadEmails: UnsignedInt,
+                   totalThreads: UnsignedInt,
+                   unreadThreads: UnsignedInt,
+                   mailboxRights: MailboxRights,
+                   isSubscribed: Boolean,
+                   namespace: MailboxNamespace,
```

Comment:
I don't see namespace in jmap spec, what did I missed?

Reply:
That's required for the James specific sharing extension

(To know if the mailbox belong to you or not)